api/v1beta1/lokistack_types.go

```diff
@@ -416,12 +423,28 @@ type IngestionLimitSpec struct {
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:com.tectonic.ui:number",displayName="Max Global Streams per  Tenant"
 	MaxGlobalStreamsPerTenant int32 `json:"maxGlobalStreamsPerTenant,omitempty"`
 
-	// MaxLineSize defines the aximum line size on ingestion path. Units in Bytes.
+	// MaxLineSize defines the maximum line size on ingestion path. Units in Bytes.
 	//
 	// +optional
 	// +kubebuilder:validation:Optional
 	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:com.tectonic.ui:number",displayName="Max Line Size"
 	MaxLineSize int32 `json:"maxLineSize,omitempty"`
+
+	// PerStreamRateLimit defines the maximum byte rate per second for each stream. Units in Bytes.
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:com.tectonic.ui:number",displayName="Per Stream Rate Limit"
+	PerStreamRateLimit int32 `json:"perStreamRateLimit,omitempty"`
+
+	// PerStreamRateLimitBurst defines the maximum burst bytes per stream.
+	// This value sets the upper limit for the stream rate limit. When exceeding
+	// this, the stream is limited.
+	//
+	// +optional
+	// +kubebuilder:validation:Optional
+	// +operator-sdk:csv:customresourcedefinitions:type=spec,xDescriptors="urn:alm:descriptor:com.tectonic.ui:number",displayName="Per Stream Rate Limit Burst"
+	PerStreamRateLimitBurst int32 `json:"perStreamRateLimitBurst,omitempty"`
```

Comment:
Why do we introduce these two limits in this PR? Can you elaborate why this is mandatory to do now? We should change the API after having a proper discussion i.e. an enhancement proposal. Adding new knobs widens our opinionated API and we should have a good reason to do so.

Reply:
Gotcha, sorry about that. Jumped a little ahead. These options were not in 2.2.1, but offer more opportunities to tune the ingestion rate of the cluster. So maybe it is worth the enhancement proposal. I will revert this change and just set their default values in the config file.